### PR TITLE
Modify rule S5000: LaYC format

### PR DESCRIPTION
**Pull request by frederic-tingaud-sonarsource**

### CODE REVIEW

File: rules/S5000/cfamily/rule.adoc [hunk 1]
```diff
@@ -1,19 +1,23 @@
-== Why is this an issue?
+The function ``++memcmp++`` only returns meaningful results for objects of trivially copyable types that have no padding. This includes scalar types, arrays, and trivially copyable classes.
```
Comment (amelie-renard-sonarsource) [inline rules/S5000/cfamily/rule.adoc:1]:
```suggestion
The function ``++memcmp++`` only returns meaningful results for objects of trivially copyable types without padding. This includes scalar types, arrays, and trivially copyable classes.
```

File: rules/S5000/cfamily/rule.adoc [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
 
-The function ``++memcmp++`` can only be used for objects of trivially copyable types. This includes scalar types, arrays, and trivially copyable classes. 
+== Why is this an issue?
 
+`memcmp` compares the raw memory representation of two objects (what the standard calls their _object representation_). For objects that are not trivially copyable or that contain padding, two objects storing identical values could have different raw memory representations, so the result of `memcmp` is not meaningful.
```
Comment (amelie-renard-sonarsource) [inline rules/S5000/cfamily/rule.adoc:5]:
```suggestion
`memcmp` compares the raw memory representation of two objects (what the standard calls their _object representation_). When objects are not trivially copyable or contain padding, they could have different raw memory representations even though they store identical values. So the result of `memcmp` is not meaningful.
```

File: rules/S5000/cfamily/rule.adoc [hunk 3]
```diff
@@ -1,19 +1,23 @@
...
 
+=== Code examples
 
-=== Noncompliant code example
+==== Noncompliant code example
 
-[source,cpp]
+[source,cpp,diff-id=1,diff-type=noncompliant]
 ----
 class Shape { // Trivially copyable, but will contain padding after the bool on most architectures
```
Comment (amelie-renard-sonarsource) [inline rules/S5000/cfamily/rule.adoc:22]: Do you think we can make the first example more C oriented?

File: rules/S5000/cfamily/rule.adoc [hunk 4]
```diff
@@ -1,19 +1,23 @@
...
 
-The function ``++memcmp++`` can only be used for objects of trivially copyable types. This includes scalar types, arrays, and trivially copyable classes. 
+== Why is this an issue?
 
+`memcmp` compares the raw memory representation of two objects (what the standard calls their _object representation_). For objects that are not trivially copyable or that contain padding, two objects storing identical values could have different raw memory representations, so the result of `memcmp` is not meaningful.
 
 A class type is trivially copyable if:
 
-* One or more of the following methods is trivial and the rest are deleted: copy constructor, move constructor, copy assignment operator, and move assignment operator,
+* One or more of the following methods is trivial and the rest are deleted: copy /constructor, move constructor, copy assignment operator, and move assignment operator,
```
Comment (amelie-renard-sonarsource) [inline rules/S5000/cfamily/rule.adoc:9]: What is a trivial copy constructor? A trivial move constructor? A trivial destructor? etc...
I think this should also be explained, don't you think?

**Review by amelie-renard-sonarsource — CHANGES_REQUESTED** [review]

File: rules/S5000/cfamily/rule.adoc (hunk 4 above)
Comment (frederic-tingaud-sonarsource) [reply rules/S5000/cfamily/rule.adoc:9]: Added explanations.

File: rules/S5000/cfamily/rule.adoc (hunk 3 above)
Comment (frederic-tingaud-sonarsource) [reply rules/S5000/cfamily/rule.adoc]: Done

File: rules/S5000/cfamily/rule.adoc [hunk 5]
```diff
@@ -1,55 +1,89 @@
...
+bool isSame(Resource *r1, Resource *r2)
 {
-    return (*s1) == (*s2);
+    return (*r1) == (*r2);
 }
 ----
 
+== Resources
+
+https://en.cppreference.com/w/cpp/language/classes#Trivially_copyable_class[Detailed definition of what a trivially copyable class is].
```
Comment (amelie-renard-sonarsource) [inline rules/S5000/cfamily/rule.adoc:86]:
```suggestion
https://en.cppreference.com/w/cpp/language/classes#Trivially_copyable_class[Definition of a trivially copyable class].
```

**Review by amelie-renard-sonarsource — APPROVED** [review]
A typo and a small suggestion, otherwise looks good!

**Pull request merged**